connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.http.base.utils;
+
+import com.google.api.client.http.GenericUrl;
+import io.camunda.connector.api.error.ConnectorInputException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class HttpBlocklistManager {
+
+  private static final String BLOCK_PREFIX = "CONNECTORS_HTTP_BLOCK_";
+  private final Map<BlockType, Set<String>> blocklist = new HashMap<>();
+  private final Map<String, Pattern> regexCache = new HashMap<>();
+
+  enum BlockType {
+    URL("URL"),
+    URL_PARTIAL_MATCH("URLPARTIAL"),
+    DOMAIN("DOMAIN"),
+    PORT("PORT"),
+    IP("IP"),
+    REGEX("REGEX");
+
+    private static final Map<String, BlockType> BY_NAME = new HashMap<>();
+
+    static {
+      for (BlockType blockType : values()) {
+        BY_NAME.put(blockType.name, blockType);
+      }
+    }
+
+    private final String name;
+
+    BlockType(String name) {
+      this.name = name;
+    }
+
+    public static BlockType valueOfName(String name) {
+      return BY_NAME.get(name);
+    }
+  }
+
+  public HttpBlocklistManager() {
+    this(System.getenv());
+  }
+
+  public HttpBlocklistManager(Map<String, String> environment) {
+    loadBlocklistFromEnv(environment);
+  }
+
+  private void loadBlocklistFromEnv(final Map<String, String> environment) {
+    for (Map.Entry<String, String> entry : environment.entrySet()) {
+      if (entry.getKey().startsWith(BLOCK_PREFIX)) {
+        String blockTypeStr = entry.getKey().substring(BLOCK_PREFIX.length()).split("_")[0];
+        Optional.ofNullable(BlockType.valueOfName(blockTypeStr))
+            .ifPresent(
+                blockType ->
+                    blocklist
+                        .computeIfAbsent(blockType, k -> new HashSet<>())
+                        .add(entry.getValue()));
+      }
+    }
+  }
```

Comment:
This method is untested. Please add a test.

Also please store the name of the block and make it available when trowing an error.
Example: `CONNECTORS_HTTP_BLOCK_URL_GOOGLE_META=metaData`
Name: `GOOGLE_META`.